test/sql/insert_single.sql

```diff
@@ -129,3 +129,26 @@ INSERT INTO "test_tz" VALUES('2017-09-21 19:01:00', 21.2);
 
 SELECT * FROM test.show_constraints('_timescaledb_internal._hyper_10_20_chunk');
 SELECT * FROM test_tz;
+
+--test rollback
+BEGIN;
+\set QUIET off
+CREATE TABLE "data_records" ("time" bigint NOT NULL, "value" integer CHECK (VALUE > 0));
+SELECT create_hypertable('data_records', 'time', chunk_time_interval => 2592000000);
+
+SAVEPOINT savepoint_1;
```

Comment:
What happens when you have nested savepoints (if possible)? (Not clear if this test covers that case.) For instance:

```
SAVEPOINT save1;
INSERT INTO "data_records" ("time", "value") VALUES (1, 1);
SAVEPOINT save2;
...
ROLLBACK TO SAVEPOINT save2;
```
  
It seems to me that, ideally, if one rolls back to a savepoint, one would also roll back the state of the cache to that point. It is not clear to me that this happens. From the code, it seems that all caches are released once a rollback happens, which seems to indicate a cache that was valid at `save1` will be invalid when rolling back to `save2`. But maybe I don't understand how this really works.

Reply:
So our cache gets fully invalidated on every rollback. But when it's repopulated it is repopulated using the proper default postgres txn rules. I don't seen why we would have to do anything different here. I added a savepoint_3 to the tests to test this some more.